crpg.py

```diff
@@ -5,60 +5,61 @@
 # the base crpg game
 
 class Game:
-    def __init__(self, player = Player(), starting_location = None):
-        self.current = starting_location
+    def __init__(self, player: Player = Player(), starting_location: Location = None):
+        self.current: Node = starting_location
         self.player = player
-        self.name = None
-
-        self.nodes = set()
-        self.connections = {}
+        self.name: str = None
 
+        self.nodes: set[Node] = set()
+        
         self.add_node(self.current)
 
     # establish nodes and connections
-
-    def add_node(self, node):
+    def add_node(self, node: Node):
         self.nodes.add(node)
 
-    def add_oneway_connection(self, node_a, node_b):
+    def add_oneway_connection(self, node_a: Node, node_b: Node):
         assert node_a in self.nodes
         assert node_b in self.nodes
 
-        # adding node_a to list of connections
-        self.connections[node_a] = self.connections.get(node_a, []) + [node_b]
+        node_a.add_connection(node_b)
```

Comment:
If we can get the node connections to work, I really like the way you exposed the API to the game